docs/source/schema_models.rst

```diff
@@ -208,6 +214,100 @@ however, a couple of gotchas:
     ...
     TypeError: Parameters to generic types must be types. Got StringDtype.
 
+.. _parameterized dtypes:
+
+Parametrized dtypes
+^^^^^^^^^^^^^^^^^^^
+Pandas supports a couple of parametrized dtypes. As of pandas 1.2.0:
+
+
++-------------------+---------------------------+-----------------------------+
+| Kind of Data      | Data Type                 | Parameters                  |
++===================+===========================+=============================+
+| tz-aware datetime | :class:`DatetimeTZDtype`  | ``unit``, ``tz``            |
++-------------------+---------------------------+-----------------------------+
+| Categorical       | :class:`CategoricalDtype` | ``categories``, ``ordered`` |
++-------------------+---------------------------+-----------------------------+
+| period            | :class:`PeriodDtype`      | ``freq``                    |
++-------------------+---------------------------+-----------------------------+
+| sparse            | :class:`SparseDtype`      | ``dtype``, ``fill_value``   |
++-------------------+---------------------------+-----------------------------+
+| intervals         | :class:`IntervalDtype`    | ``subtype``                 |
++-------------------+---------------------------+-----------------------------+
+
+Annotated
+"""""""""
+
+Parameters can be given via :data:`typing.Annotated`. It requires python > 3.9 or
+`typing_extensions <https://pypi.org/project/typing-extensions/>`_, which is already a
+requirement of Pandera. Unfortunately :data:`typing.Annotated` has not been backported
+to python 3.6.
+
+:green:`✔` Good:
+
+.. testcode:: dataframe_schema_model
+    :skipif: PY36
+
+    import sys
+
+    if (3, 6) < sys.version_info[:2] < (3, 9):
```

Comment:
consider something a little more concise:
```python
# python 3.7+
try:
    from typing import Annotated
except:
    from typing_extensions import Annotated
```

The above paragraph establishes that this won't work for python 3.6, so I think it'd be understood by the reader that this code block only applies to 3.7+

Reply:
Good catch, your suggestion looks nicer :+1:

Reply:
Recently, pandas 1.2.0 [dropped support for python 3.6](https://pandas.pydata.org/docs/whatsnew/v1.2.0.html#increased-minimum-version-for-python). Pandera could follow suit and require pandas 1.2.0 + py3.7. 

Pandera's api is still maturing, with new features added every couple of months. I think users should expect a young library to move fast. Pandas 1.0 is a major version and should be preferred in production environment using modern tooling like pandera.